gateway-service/src/main/java/org/zowe/apiml/gateway/security/config/NewSecurityConfiguration.java

```diff
@@ -0,0 +1,394 @@
+                .userDetailsService(new SimpleUserDetailService())
+                .and()
+                .addFilterBefore(ticketFilter("/**", authenticationManager()), UsernamePasswordAuthenticationFilter.class);
+        }
+
+        private QueryFilter ticketFilter(String ticketEndpoint, AuthenticationManager authenticationManager) {
+            return new QueryFilter(
+                ticketEndpoint,
+                successfulTicketHandler,
+                handlerInitializer.getAuthenticationFailureHandler(),
+                authenticationService,
+                HttpMethod.POST,
+                true,
+                authenticationManager);
+        }
+    }
+
+    @Configuration
+    @RequiredArgsConstructor
+    @Order(4)
+    class CertificateProtectedEndpoints extends WebSecurityConfigurerAdapter {
+
+        @Override
+        protected void configure(HttpSecurity http) throws Exception {
+            baseConfigure(http.requestMatchers()
+                    .antMatchers(HttpMethod.DELETE,CacheServiceController.CONTROLLER_PATH + "/**")
+                    .antMatchers(AuthController.CONTROLLER_PATH + AuthController.INVALIDATE_PATH, AuthController.CONTROLLER_PATH + AuthController.DISTRIBUTE_PATH).and()
+                ).authorizeRequests()
+                .anyRequest().authenticated()
+                .and()
+                .logout().disable() // logout filter in this chain not needed
+                .x509() // default x509 filter, authenticates trusted cert
+                .subjectPrincipalRegex(EXTRACT_USER_PRINCIPAL_FROM_COMMON_NAME)
+                .userDetailsService(new SimpleUserDetailService());
+        }
+    }
+
+    @Configuration
+    @RequiredArgsConstructor
+    @Order(5)
+    class CertificateOrAuthProtectedEndpoints extends WebSecurityConfigurerAdapter {
+
+        private final CompoundAuthProvider compoundAuthProvider;
+        private final AuthenticationProvider tokenAuthenticationProvider;
+
+        private final String[] protectedEndpoints = {
+            "/application",
+            ServicesInfoController.SERVICES_URL
+        };
+
+        @Override
+        protected void configure(AuthenticationManagerBuilder auth) {
+            auth.authenticationProvider(compoundAuthProvider); // for authenticating credentials
+            auth.authenticationProvider(tokenAuthenticationProvider); // for authenticating Tokens
+        }
+
+        @Override
+        protected void configure(HttpSecurity http) throws Exception {
+            baseConfigure(http.requestMatchers()
+                .antMatchers("/application/**")
+                .antMatchers(ServicesInfoController.SERVICES_URL + "/**").and()
+            ).authorizeRequests()
+                .anyRequest().authenticated()
+                .and()
+                .logout().disable() // logout filter in this chain not needed
+                .x509() // default x509 filter, authenticates trusted cert

```

Comment:
Again same concern around routing to endpoint via Gateway.

Reply:
Calling things through Gateway with certificate is a topic that is encapsulated in HttpClientChooser class. It's not related to gateway's filterchain. Do you have specific concern in mind?
There is one concern here for the extender that raised #1523. I have discussed the limitations with him and he is ok with the implementation, as noted in the issue.